centreon/src/Core/ResourceAccess/Application/UseCase/UpdateRule/UpdateRuleValidation.php

```diff
@@ -0,0 +1,137 @@
+
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Contact\Application\Repository\ReadContactGroupRepositoryInterface;
+use Core\Contact\Application\Repository\ReadContactRepositoryInterface;
+use Core\ResourceAccess\Application\Exception\RuleException;
+use Core\ResourceAccess\Application\Providers\DatasetProviderInterface;
+use Core\ResourceAccess\Application\Repository\ReadResourceAccessRepositoryInterface;
+use Core\ResourceAccess\Domain\Model\NewRule;
+
+class UpdateRuleValidation
+{
+    use LoggerTrait;
+
+    /** @var DatasetProviderInterface[] */
+    private array $repositoryProviders;
+
+    /**
+     * @param ReadResourceAccessRepositoryInterface $repository
+     * @param ReadContactRepositoryInterface $contactRepository
+     * @param ReadContactGroupRepositoryInterface $contactGroupRepository
+     * @param \Traversable<DatasetProviderInterface> $repositoryProviders
+     */
+    public function __construct(
+        private readonly ReadResourceAccessRepositoryInterface $repository,
+        private readonly ReadContactRepositoryInterface $contactRepository,
+        private readonly ReadContactGroupRepositoryInterface $contactGroupRepository,
+        \Traversable $repositoryProviders
+    ) {
+        $this->repositoryProviders = iterator_to_array($repositoryProviders);
+    }
+
+    /**
+     * Validates that the name provided for the rule is not already used.
+     *
+     * @param string $name
+     *
+     * @throws RuleException
+     */
+    public function assertIsValidName(string $name): void
+    {
+        $this->debug('Check that resource access rule name is not already used', ['name' => $name]);
+        if ($this->repository->existsByName(NewRule::formatName($name))) {
+            $this->error('Resource access rule name already used', ['name' => $name]);
+
+            throw RuleException::nameAlreadyExists(NewRule::formatName($name), $name);
+        }
+    }
+
+    /**
+     * @param int[] $contactIds
+     *
+     * @throws RuleException
+     */
+    public function assertContactIdsAreValid(array $contactIds): void
+    {
+        $validIds = $this->contactRepository->exist($contactIds);
+
+        if ([] !== ($invalidIds = array_diff($contactIds, $validIds))) {
+            throw RuleException::idsDoNotExist('contactIds', $invalidIds);
+        }
+    }
+
+    /**
+     * @param int[] $contactGroupIds
+     *
+     * @throws RuleException
+     */
+    public function assertContactGroupIdsAreValid(array $contactGroupIds): void
+    {
+        $validIds = $this->contactGroupRepository->exist($contactGroupIds);
+
+        if ([] !== ($invalidIds = array_diff($contactGroupIds, $validIds))) {
+            throw RuleException::idsDoNotExist('contactGroupIds', $invalidIds);
+        }
+    }
+
+    /**
+     * @param string $type
+     * @param int[] $ids
+     *
+     * @throws RuleException
+     */
+    public function assertIdsAreValid(string $type, array $ids): void
```

Comment:
Ids of ?

Reply:
done ✅ renamed to `assertResourceIdsAreValid`